ibis/backends/polars/__init__.py

```diff
@@ -72,7 +72,11 @@ def list_tables(self, like=None, database=None):
         return self._filter_with_like(list(self._tables.keys()), like)
 
     def table(self, name: str) -> ir.Table:
-        schema = sch.infer(self._tables[name])
+        try:
+            schema = sch.infer(self._tables[name])
+        except KeyError:
+            raise com.TableNotFound(name)
```

Comment:
```suggestion
        try:
            table = self._tables[name]
        except KeyError:
            raise com.TableNotFound(name)
            
        schema = sch.infer(table)
```

Otherwise, we're potentially catching bugs from `KeyError`s raised in `sch.infer`, which we definitely cannot guarantee won't happen.

Generally speaking, making a single call in a `try:` block is preferred, especially if the exception type you're catching is extremely common or generic.

Reply:
I see your point but, there is a problem with this as `self._tables[name]` doesn't raise the `KeyError`. it's 
`self._tables[name]` is `<LazyFrame at 0x12C92B250>` it only raises once we try to infer the schema, so the workflow suggested doesn't work. Not sure what's the best way to do this here